src/data/matrix/hadamard.lean

```diff
@@ -0,0 +1,142 @@
+/-
+Copyright (c) 2021 Lu-Ming Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Lu-Ming Zhang
+-/
+import data.matrix.notation
+import linear_algebra.matrix.trace
+import data.complex.basic
+
+/-!
+# Hadamard product of matrices
+
+This file defines the Hadamard product `matrix.Hadamard`
+and contains basic properties about them.
+
+## Main definition
+
+- `matrix.Hadamard`: defines the Hadamard product,
+  which is the pointwise product of two matrices of the same size.
+
+## Notation
+
+* `⊙`: the Hadamard product `matrix.Hadamard`;
+
+## References
+
+*  <https://en.wikipedia.org/wiki/Hadamard_product_(matrices)>
+
+## Tags
+
+Hadamard product, Hadamard
+-/
+
+variables {α β γ I J K L M N: Type*}
+variables {R : Type*}
+variables {m n p q r s t: ℕ}
+
+namespace matrix
+open_locale matrix big_operators
+open complex
+
+/-- `matrix.Hadamard` defines the Hadamard product,
+    which is the pointwise product of two matrices of the same size.-/
+def Hadamard [has_mul α] (A : matrix I J α) (B : matrix I J α) :
+matrix I J α :=
+λ i j, (A i j) * (B i j)
```

Comment:
```suggestion
@[simp]
def Hadamard [has_mul α] (A : matrix I J α) (B : matrix I J α) : matrix I J α
| i j := A i j * B i j
```
tells lean "whenever you see `Hadamard A B i j` unfold it". Note that this is not the same as putting `@[simp]` on what you already have, which would tell lean to unfold `Hadamard A B` which is undesirable.